### PR TITLE
Create GitHub Actions to automate Release Note author update

### PR DESCRIPTION
**Pull request by yongtang**
This PR create GitHub Actions to automate Release Note author update. Anytime a commit is made, the bot will automatically update the PR to make sure the Release note captures the author (if new author is available).

Note the bot will compare the version in tensorflow_io/python/ops/version_ops.py and the RELEASE.md's head version, and will create a placeholder section if version_ops.py is later than RELEASE.md's latest version.

Example PR is available: https://github.com/yongtang/io/pull/12

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

File: tools/release/note_update.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    if re.match(r"^# Release [\d\.]+", line):
+        entries.append((line[len("# Release ") :].rstrip(), index))
+
+# Find current version
+curr = tuple(map(int, (entries[0][0].split("."))))
+# Find last version, padded to patch version of 0
+if curr[2] == 0:
+    last = tuple([curr[0], curr[1] - 1, 0])
+else:
+    last = tuple([curr[0], curr[1], 0])
```
Comment (kvignesh1420) [inline tools/release/note_update.py:19]: I think it should be:
```python
else:
    last = tuple([curr[0], curr[1], curr[2]-1])
```

**Review by kvignesh1420 — COMMENTED** [review]
Also, just as a follow-up, I think it would be beneficial to use [semver](https://github.com/python-semver/python-semver) and bump the major/minor version seamlessly. WDYT? I will take a look after this PR is merged.

File: tools/release/note_update.py (hunk 1 above)
Comment (yongtang) [reply tools/release/note_update.py]: @kvignesh1420 PR updated.

**Comment by yongtang** [comment]
> Also, just as a follow-up, I think it would be beneficial to use [semver](https://github.com/python-semver/python-semver) and bump the major/minor version seamlessly. WDYT? I will take a look after this PR is merged.

@kvignesh1420 sounds good to me 👍

**Review by kvignesh1420 — APPROVED** [review]

**Pull request merged**